### PR TITLE
Add xml comments for introspection types

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 1]
```diff
@@ -3,40 +3,52 @@
 
 namespace GraphQL.Introspection
 {
+    /// <summary>
+    /// Represents an filter for information returned from introspection queries.
+    /// Can be used to hide information, such as graph types, from clients that have an inadequate permission level.
+    /// </summary>
     public interface ISchemaFilter
```
Comment (sungam3r) [inline src/GraphQL/Introspection/ISchemaFilter.cs:10]: We already have comments for this file in `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 2]
```diff
@@ -3,40 +3,52 @@
 
 namespace GraphQL.Introspection
 {
+    /// <summary>
+    /// Represents an filter for information returned from introspection queries.
+    /// Can be used to hide information, such as graph types, from clients that have an inadequate permission level.
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:8]:
```suggestion
    /// Provides the ability to filter the schema upon introspection to hide types, fields, arguments, enum values, and directives.
    /// Can be used to hide information, such as graph types, from clients that have an inadequate permission level.
```
Updated from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 3]
```diff
@@ -3,40 +3,52 @@
...
+    /// </summary>
     public interface ISchemaFilter
     {
+        /// <summary>
+        /// Returns a boolean indicating whether the specified graph type should be returned within the introspection query.
+        /// </summary>
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:14]:
```suggestion
        /// <summary>
        /// Returns a boolean indicating whether the specified graph type should be returned within the introspection query.
        /// </summary>
        /// <param name="type">The graph type to consider.</param>
```
Updated from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 4]
```diff
@@ -3,40 +3,52 @@
...
         Task<bool> AllowType(IGraphType type);
+        /// <summary>
+        /// Returns a boolean indicating whether the specified field should be returned within the introspection query.
+        /// </summary>
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:18]:
```suggestion
        /// <summary>
        /// Returns a boolean indicating whether the specified field should be returned within the introspection query.
        /// </summary>
        /// <param name="parent">The parent type to which the field belongs.</param>
        /// <param name="field">The field to consider.</param>
```
Updated from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 5]
```diff
@@ -3,40 +3,52 @@
...
         Task<bool> AllowField(IGraphType parent, IFieldType field);
+        /// <summary>
+        /// Returns a boolean indicating whether the specified argument should be returned within the introspection query.
+        /// </summary>
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:22]:
```suggestion
        /// <summary>
        /// Returns a boolean indicating whether the specified argument should be returned within the introspection query.
        /// </summary>
        /// <param name="field">The field to which the argument belongs.</param>
        /// <param name="argument">The argument to consider.</param>
```

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 6]
```diff
@@ -3,40 +3,52 @@
...
         Task<bool> AllowArgument(IFieldType field, QueryArgument argument);
+        /// <summary>
+        /// Returns a boolean indicating whether the specified enumeration value should be returned within the introspection query.
+        /// </summary>
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:26]:
```suggestion
        /// <summary>
        /// Returns a boolean indicating whether the specified enumeration value should be returned within the introspection query.
        /// </summary>
        /// <param name="parent">The enumeration to which the enumeration value belongs.</param>
        /// <param name="enumValue">The enumeration value to consider.</param>
```
Updated from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 7]
```diff
@@ -3,40 +3,52 @@
...
         Task<bool> AllowEnumValue(EnumerationGraphType parent, EnumValueDefinition enumValue);
+        /// <summary>
+        /// Returns a boolean indicating whether the specified directive should be returned within the introspection query.
+        /// </summary>
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:30]:
```suggestion
        /// <summary>
        /// Returns a boolean indicating whether the specified directive should be returned within the introspection query.
        /// </summary>
        /// <param name="directive">The directive to consider.</param>
```
Updated from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 8]
```diff
@@ -3,40 +3,52 @@
...
         Task<bool> AllowDirective(DirectiveGraphType directive);
     }
 
+    /// <summary>
+    /// A schema filter which allows all requested information to introspection queries.
+    /// </summary>
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:36]:
```suggestion
    /// <summary>
    /// The default schema filter.  By default nothing is hidden.
    /// </summary>
```
Updated from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 9]
```diff
@@ -3,40 +3,52 @@
...
     public class DefaultSchemaFilter : ISchemaFilter
     {
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:38]:
```suggestion
    {
        private static readonly Task<bool> _completed = Task.FromResult(true);

```
Updated from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 10]
```diff
@@ -3,40 +3,52 @@
...
-        public virtual Task<bool> AllowType(IGraphType type)
-        {
-            return Task.FromResult(true);
-        }
+        /// <inheritdoc/>
+        public virtual Task<bool> AllowType(IGraphType type) => Task.FromResult(true);
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:40]:
```suggestion
        public virtual Task<bool> AllowType(IGraphType type) => _completed;
```

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 11]
```diff
@@ -3,40 +3,52 @@
...
 
-        public virtual Task<bool> AllowField(IGraphType parent, IFieldType field)
-        {
-            return Task.FromResult(true);
-        }
+        /// <inheritdoc/>
+        public virtual Task<bool> AllowField(IGraphType parent, IFieldType field) => Task.FromResult(true);
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:43]:
```suggestion
        public virtual Task<bool> AllowField(IGraphType parent, IFieldType field) => _completed;
```

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 12]
```diff
@@ -3,40 +3,52 @@
...
 
-        public virtual Task<bool> AllowArgument(IFieldType field, QueryArgument argument)
-        {
-            return Task.FromResult(true);
-        }
+        /// <inheritdoc/>
+        public virtual Task<bool> AllowArgument(IFieldType field, QueryArgument argument) => Task.FromResult(true);
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:46]:
```suggestion
        public virtual Task<bool> AllowArgument(IFieldType field, QueryArgument argument) => _completed;
```

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 13]
```diff
@@ -3,40 +3,52 @@
...
 
-        public virtual Task<bool> AllowEnumValue(EnumerationGraphType parent, EnumValueDefinition enumValue)
-        {
-            return Task.FromResult(true);
-        }
+        /// <inheritdoc/>
+        public virtual Task<bool> AllowEnumValue(EnumerationGraphType parent, EnumValueDefinition enumValue) => Task.FromResult(true);
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:49]:
```suggestion
        public virtual Task<bool> AllowEnumValue(EnumerationGraphType parent, EnumValueDefinition enumValue) => _completed;
```

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 14]
```diff
@@ -3,40 +3,52 @@
...
 
-        public virtual Task<bool> AllowDirective(DirectiveGraphType directive)
-        {
-            return Task.FromResult(true);
-        }
+        /// <inheritdoc/>
+        public virtual Task<bool> AllowDirective(DirectiveGraphType directive) => Task.FromResult(true);
```
Comment (Shane32) [inline src/GraphQL/Introspection/ISchemaFilter.cs:52]:
```suggestion
        public virtual Task<bool> AllowDirective(DirectiveGraphType directive) => _completed;
```

**Review by Shane32 — COMMENTED** [review]
Updates backported from `develop`.

File: src/GraphQL/Introspection/ISchemaFilter.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL/Introspection/ISchemaFilter.cs:10]: I've backported/merged the comments from `develop` back here.  Please re-review.

File: src/GraphQL/Introspection/ISchemaFilter.cs [hunk 15]
```diff
@@ -3,40 +3,62 @@
...
         Task<bool> AllowEnumValue(EnumerationGraphType parent, EnumValueDefinition enumValue);
+        /// <summary>
+        /// Returns a boolean indicating whether the specified directive should be returned within the introspection query.
+        /// </summary>
+        /// <param name="directive">The directive to consider.</param>
         Task<bool> AllowDirective(DirectiveGraphType directive);
     }
 
+    /// <summary>
+    /// The default schema filter.  By default nothing is hidden.
```
Comment (sungam3r) [inline src/GraphQL/Introspection/ISchemaFilter.cs:43]:
```suggestion
    /// The default schema filter. By default nothing is hidden.
```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**